tests/pytests/unit/test_minion.py

```diff
@@ -21,35 +21,33 @@
 log = logging.getLogger(__name__)
 
 
-def test_minion_load_grains_false():
+def test_minion_load_grains_false(minion_opts):
     """
     Minion does not generate grains when load_grains is False
     """
-    opts = {"random_startup_delay": 0, "grains": {"foo": "bar"}}
+    minion_opts["grains"] = {"foo": "bar"}
     with patch("salt.loader.grains") as grainsfunc:
-        minion = salt.minion.Minion(opts, load_grains=False)
-        assert minion.opts["grains"] == opts["grains"]
+        minion = salt.minion.Minion(minion_opts, load_grains=False)
+        assert minion.opts["grains"] == minion_opts["grains"]
         grainsfunc.assert_not_called()
 
 
-def test_minion_load_grains_true():
+def test_minion_load_grains_true(minion_opts):
     """
     Minion generates grains when load_grains is True
     """
-    opts = {"random_startup_delay": 0, "grains": {}}
```

Comment:
Can you elaborate on why you are removing these opts from this test and the ones below?

Reply:
minion_opts is a dict with those keys and values init

Reply:
Yesh, but are these tests relying on non default configs? For example in a test below it had previously set `"return_retry_tries"` to 20 which is not the default.

Reply:
yup line 148 changes the default to 20